docroot/themes/humsci/humsci_basic/src/scss/utilities/_color-pairings.scss

```diff
@@ -1,55 +1,22 @@
+// The following are the classes that update the CSS --palette variables.
+// They should be applied to a parent element.
+// Note: If a utility class is not specified colors will fallback to the defaults
+// defined in elements/_base.scss
+
 // Generate Color Palette Classes for Colorful
-@each $palette, $color-swatches in $hc-colorful-pairings {
-  // These are the classes that update the CSS --palette variable.
-  // They should be applied to the parent element.
-  // Note: If a hc-pairing- class is not applied, hc-pairing-ocean will apply as the default / fallback color pairing.
-  .hc-pairing-#{$palette} {
-    // BASE
-    --palette--primary: #{hb-get-pairing-color("primary", $palette, $hc-colorful-pairings)};
-    --palette--secondary: #{hb-get-pairing-color("secondary", $palette, $hc-colorful-pairings)};
-    --palette--tertiary: #{hb-get-pairing-color("tertiary", $palette, $hc-colorful-pairings)};
-    // HERO
-    --palette--primary-hero-overlay: #{hb-get-pairing-color("primary-hero-overlay", $palette, $hc-colorful-pairings)};
-    // PRIMARY VARIANTS
-    --palette--primary-dark: #{hb-get-pairing-color("primary-dark", $palette, $hc-colorful-pairings)};
-    // SECONDARY VARIANTS
-    --palette--secondary-active: #{hb-get-pairing-color("secondary-active", $palette, $hc-colorful-pairings)};
-    --palette--secondary-highlight: #{hb-get-pairing-color("secondary-highlight", $palette, $hc-colorful-pairings)};
-    --palette--secondary-highlight-darken: #{hb-get-pairing-color("secondary-highlight-darken", $palette, $hc-colorful-pairings)};
-    --palette--secondary-darken-12: #{hb-get-pairing-color("secondary-darken-12", $palette, $hc-colorful-pairings)};
-    // TERTIARY VARIANTS
-    --palette--tertiary-highlight: #{hb-get-pairing-color("tertiary-highlight", $palette, $hc-colorful-pairings)};
-    --palette--tertiary-highlight-darken-10: #{hb-get-pairing-color("tertiary-highlight-darken-10", $palette, $hc-colorful-pairings)};
-    --palette--tertiary-reversed: #{hb-get-pairing-color("tertiary-reversed", $palette, $hc-colorful-pairings)};
-    --palette--tertiary-reversed-darken-10: #{hb-get-pairing-color("tertiary-reversed-darken-10", $palette, $hc-colorful-pairings)};
-    --palette--tertiary-darken-20: #{hb-get-pairing-color("tertiary-darken-20", $palette, $hc-colorful-pairings)};
+@include hb-themes(('colorful', 'airy')) {
+  @each $palette, $color-swatches in $hc-colorful-pairings {
```

Comment:
do you need the $color-swatches variable here since you don't use it?

Reply:
Yeah, when looping through a scss map you need to specify both they key and value or else it grabs the data differently, it'll pass both the key and value for `$palette` var.

Reply:
oo interesting. thanks!